packages/runtime/container-runtime/src/summary/summaryGenerator.ts

```diff
@@ -338,9 +338,6 @@ export class SummaryGenerator {
 		} catch (error) {
 			return fail("submitSummaryFailure", error);
 		} finally {
-			if (summaryData === undefined) {
-				this.heuristicData.recordAttempt();
```

Comment:
I don't see an equivalent to this. This means that this code is dependent on the submitSummaryCallback to always do the right thing by calling back to record an attempt. That's not the right approach and increases the chances to bugs.